### PR TITLE
Updates for 0.12

### PR DESCRIPTION
**Pull request by paulyoung**
There are a few things to discuss here, which I'll point out.

~~I think the biggest change is using [`sharkdp/purescript-decimals`](https://github.com/sharkdp/purescript-decimals) instead of [`purescript-contrib/purescript-precise`](https://github.com/purescript-contrib/purescript-precise). The former seems better maintained but has a JavaScript dependency.~~

### CODE REVIEW

File: src/Text/Markdown/SlamDown/Syntax/TextBox.purs [hunk 1]
```diff
@@ -60,23 +62,23 @@ traverseTextBox eta = case _ of
...
     PlainText def → "(PlainText " <> show def <> ")"
     Numeric def → "(Numeric " <> show def <> ")"
     Date def → "(Date " <> show def <> ")"
     Time prec def → "(Time " <> show prec <> " " <> show def <> ")"
     DateTime prec def → "(DateTime " <> show prec <> " " <> show def <> ")"
 
-derive instance eqTextBox ∷ (Functor f, Eq (f String), Eq (f HN.HugeNum), Eq (f DT.Time), Eq (f DT.Date), Eq (f DT.DateTime)) ⇒ Eq (TextBox f)
-derive instance ordTextBox ∷ (Functor f, Ord (f String), Ord (f HN.HugeNum), Ord (f DT.Time), Ord (f DT.Date), Ord (f DT.DateTime)) ⇒ Ord (TextBox f)
+derive instance eqTextBox ∷ (Functor f, Eq (f String), Eq (f D.Decimal), Eq (f DT.Time), Eq (f DT.Date), Eq (f DT.DateTime), Eq1 f) ⇒ Eq (TextBox f)
+derive instance ordTextBox ∷ (Functor f, Ord (f String), Ord (f D.Decimal), Ord (f DT.Time), Ord (f DT.Date), Ord (f DT.DateTime), Ord1 f) ⇒ Ord (TextBox f)
```
Comment (paulyoung) [inline src/Text/Markdown/SlamDown/Syntax/TextBox.purs]: I don't understand this but I had to add `Eq1` and `Ord1` here to get this to type check.

I'm seeing this test failure locally:

```
Original:
   (Right (SlamDown ((Paragraph ((FormField "name" false (TextBox (PlainText (Compose (Just (Literal "Phil Freeman")))))) : Nil)) : Nil)))
Printed:
   (Right "name  = ______ (Phil Freeman)\n")
Parsed:
   (Right (SlamDown ((Paragraph ((FormField "name" false (TextBox (PlainText (Compose (Just (Literal "Phil Freeman")))))) : Nil)) : Nil)))
/Users/py/projects/slamdata/purescript-markdown/output/Data.Eq/index.js:13
var Eq1 = function (eq1) {
                   ^

RangeError: Maximum call stack size exceeded
    at new Eq1 (/Users/py/projects/slamdata/purescript-markdown/output/Data.Eq/index.js:13:20)
    at /Users/py/projects/slamdata/purescript-markdown/output/Data.Functor.Compose/index.js:90:16
    at /Users/py/projects/slamdata/purescript-markdown/output/Data.Functor.Compose/index.js:84:66
    at /Users/py/projects/slamdata/purescript-markdown/output/Data.Functor.Compose/index.js:91:59
    at /Users/py/projects/slamdata/purescript-markdown/output/Data.Functor.Compose/index.js:84:77
    at /Users/py/projects/slamdata/purescript-markdown/output/Data.Functor.Compose/index.js:91:59
    at /Users/py/projects/slamdata/purescript-markdown/output/Data.Functor.Compose/index.js:84:77
    at /Users/py/projects/slamdata/purescript-markdown/output/Data.Functor.Compose/index.js:91:59
    at /Users/py/projects/slamdata/purescript-markdown/output/Data.Functor.Compose/index.js:84:77
    at /Users/py/projects/slamdata/purescript-markdown/output/Data.Functor.Compose/index.js:91:59
* ERROR: Subcommand terminated with exit code 1
```

Comment (paulyoung) [reply src/Text/Markdown/SlamDown/Syntax/TextBox.purs]: Well, not a _test failure_ 😄

**Comment by paulyoung** [comment]
FWIW I opened https://github.com/purescript-contrib/purescript-precise/pull/13 so we could keep using that if necessary.

File: src/Text/Markdown/SlamDown/Syntax/TextBox.purs [hunk 2]
```diff
@@ -68,8 +70,8 @@ instance showTextBox ∷ (Functor f, Show (f String), Show (f HN.HugeNum), Show
     Time prec def → "(Time " <> show prec <> " " <> show def <> ")"
     DateTime prec def → "(DateTime " <> show prec <> " " <> show def <> ")"
 
-derive instance eqTextBox ∷ (Functor f, Eq (f String), Eq (f HN.HugeNum), Eq (f DT.Time), Eq (f DT.Date), Eq (f DT.DateTime)) ⇒ Eq (TextBox f)
-derive instance ordTextBox ∷ (Functor f, Ord (f String), Ord (f HN.HugeNum), Ord (f DT.Time), Ord (f DT.Date), Ord (f DT.DateTime)) ⇒ Ord (TextBox f)
+derive instance eqTextBox ∷ (Functor f, Eq (f String), Eq (f HN.HugeNum), Eq (f DT.Time), Eq (f DT.Date), Eq (f DT.DateTime), Eq1 f) ⇒ Eq (TextBox f)
```
Comment (cryogenian) [inline src/Text/Markdown/SlamDown/Syntax/TextBox.purs]: Maybe this should be 
```purescript
derive instance eqTextBox :: (Functor f, Eq1 f) => Eq (TextBox f) 
derive instance ordTextBox :: (Functor f, Ord1 f) => Ord (TextBox f) 
```

**Comment by cryogenian** [comment]
The fact travis is unhappy is very interesting, I'm not absolutely sure, but this might be somehow connected with `Eq1` deriving. Perhaps @garyb could say something about it :)

File: src/Text/Markdown/SlamDown/Syntax/TextBox.purs [hunk 3]
```diff
@@ -68,8 +70,8 @@ instance showTextBox ∷ (Functor f, Show (f String), Show (f HN.HugeNum), Show
     Time prec def → "(Time " <> show prec <> " " <> show def <> ")"
     DateTime prec def → "(DateTime " <> show prec <> " " <> show def <> ")"
 
-derive instance eqTextBox ∷ (Functor f, Eq (f String), Eq (f HN.HugeNum), Eq (f DT.Time), Eq (f DT.Date), Eq (f DT.DateTime)) ⇒ Eq (TextBox f)
-derive instance ordTextBox ∷ (Functor f, Ord (f String), Ord (f HN.HugeNum), Ord (f DT.Time), Ord (f DT.Date), Ord (f DT.DateTime)) ⇒ Ord (TextBox f)
+derive instance eqTextBox ∷ (Functor f, Eq1 f) ⇒ Eq (TextBox f)
+derive instance ordTextBox ∷ (Functor f, Ord1 f) ⇒ Ord (TextBox f)
```
Comment (paulyoung) [inline src/Text/Markdown/SlamDown/Syntax/TextBox.purs]: I believe this is the code relating to the stack overflow.

**Comment by garyb** [comment]
I'll take a look!

**Comment by garyb** [comment]
Ok, after much head scratching, it's a problem with `Eq` for `Compose`: https://github.com/purescript/purescript-functors/pull/16

I'll push some other changes into this branch though, as a lot of the `Eq` / `Ord` constraints in here are avoidable.

**Comment by paulyoung** [comment]
I think this should be good now. Thanks @garyb!

**Pull request merged**